server/ec2spotmanager/CloudProvider/EC2SpotCloudProvider.py

```diff
@@ -0,0 +1,379 @@
+                                    pool_id, region, msg)
+                raise Exception({"type": "temporary-failure", "data": "Temporary failure occured: %s" % str(msg)})
+
+            except Exception as msg:
+                self.logger.exception("[Pool %d] start_pool_instances: laniakea failure: %s", pool_id, msg)
+                raise Exception({"type": "unclassified", "data": str(msg)})
+
+            try:
+                instances = {}
+                self.logger.info("[Pool %d] Creating %dx %s instances... (%d cores total)", pool_id, count,
+                                 instance_type, count * CORES_PER_INSTANCE[instance_type])
+                for ec2_request in cluster.create_spot_requests(config.max_price * CORES_PER_INSTANCE[instance_type],
+                                                                delete_on_termination=True,
+                                                                timeout=10 * 60):
+                    instances[ec2_request] = {}
+                    instances[ec2_request]['instance_id'] = ec2_request
+                    instances[ec2_request]['region'] = region
+                    instances[ec2_request]['zone'] = zone
+                    instances[ec2_request]['status_code'] = INSTANCE_STATE["requested"]
+                    instances[ec2_request]['pool_id'] = pool_id
+                    instances[ec2_request]['size'] = CORES_PER_INSTANCE[instance_type]
+
+                return instances
+
+            except (boto.exception.EC2ResponseError, boto.exception.BotoServerError, ssl.SSLError, socket.error) as msg:
+                if "MaxSpotInstanceCountExceeded" in str(msg):
+                    self.logger.warning("[Pool %d] start_pool_instances: Maximum instance count exceeded for region %s",
+                                        pool_id, region)
+                    raise Exception({"type": "max-spot-instance-count-exceeded",
+                                     "data": "Auto-selected region exceeded its maximum spot instance count."})
+                elif "Service Unavailable" in str(msg):
+                    self.logger.warning("[Pool %d] start_pool_instances: Temporary failure in region %s: %s",
+                                        pool_id, region, msg)
+                    raise Exception({"type": "temporary-failure", "data": "Temporary failure occurred: %s" % str(msg)})
+                else:
+                    self.logger.exception("[Pool %d] start_pool_instances: boto failure: %s", pool_id, msg)
+                    raise Exception({"type": "unclassified", "data": "Unclassified error occurred: %s" % str(msg)})
+        except Exception as msg:
+            self.logger.exception("[Pool %d] start_pool_instances: unhandled failure: %s", pool_id, msg)
+            raise Exception({"type": "unclassified", "data": "Unclassified error occurred: %s" % str(msg)})
+
+    def check_instances_requests(self, region, pool_id, instances, tags):
```

Comment:
`instances` should be `req_ids` to match the interface.

Reply:
fixed.